File: solutions/elixir/01-gg4/code/lib/main.ex
defmodule CLI do
  def main(args) do
    command = List.first(args)
    case command do
      "init" ->
        File.mkdir!(".git")
        File.mkdir!(".git/objects")
        File.mkdir!(".git/refs")
        File.write!(".git/HEAD", "ref: refs/heads/main\n")
        IO.puts("Initialized git directory")

      _ ->
        raise "Unknown command #{command}"
    end
  end
end

File: solutions/php/01-gg4/code/app/main.php
<?php
error_reporting(E_ALL);

if (count($argv) < 2) {
    throw new RuntimeException("No command provided");
}

$command = $argv[1];
if ($command === "init") {
    mkdir(".git");
    mkdir(".git/objects");
    mkdir(".git/refs");
    file_put_contents(".git/HEAD", "ref: refs/heads/main\n");
    fwrite(STDOUT, "Initialized git directory\n");
    return;
}

throw new RuntimeException("Unknown command #{$command}");